src/main/java/seedu/address/storage/JsonAppointmentBookStorage.java

```diff
@@ -21,16 +21,27 @@
 
     private static final Logger logger = LogsCenter.getLogger(JsonAppointmentBookStorage.class);
 
+    private AppointmentArchive csvArchive;
+
     private Path filePath;
 
-    public JsonAppointmentBookStorage(Path filePath) {
+    /**
+     * @param filePath the filePath for the appointment storage.
+     * @param archivePath the directory path for the appointment archives.
+     */
+    public JsonAppointmentBookStorage(Path filePath, Path archivePath) {
         this.filePath = filePath;
+        this.csvArchive = new CsvAppointmentArchive(archivePath);
     }
 
     public Path getAppointmentBookFilePath() {
         return filePath;
     }
 
+    public Path getAppointmentArchiveDirPath() {
```

Comment:
Is a Javadoc missing here? or it is too obvious?

Reply:
Added! 👍